xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/macro/MacroId.java

```diff
@@ -106,15 +106,14 @@ public int hashCode()
     @Override
     public boolean equals(Object object)
     {
-        if (object == null) {
-            return false;
-        }
         if (object == this) {
             return true;
         }
-        if (object.getClass() != getClass()) {
-            return false;
+        
+        if (!(object instanceof MacroId)) {             // Can't be null if instance of MacroId.
```

Comment:
I don't think those comments are really required. If you absolutely want to put them put them before the if and not on the same line.

Reply:
Okay removed them